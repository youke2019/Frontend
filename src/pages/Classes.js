import React from "react";
import {View, StyleSheet, Text, ScrollView, TouchableOpacity} from "react-native";
import { Col, Row, Grid } from "react-native-easy-grid";
import Picker from 'react-native-picker'
import { loadData } from '../utils/LocalStorage'
import { getWeekClassTable } from '../utils/ClassLogics'


class Classes extends React.Component {
    state = {
        classes: [[],[],[],[],[],[],[]],
        week: 1
    }

    componentWillMount() {
        this.loadLessons()
    }


    timePick = () => {
        const range = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16];
        Picker.init({
            pickerTitleText:'上课周选择',
            pickerCancelBtnText:'取消',
            pickerConfirmBtnText:'确定',
            pickerData: range,
            selectedValue: [this.state.week],
            onPickerConfirm: data => {
                this.setState({
                    week: parseInt(data[0])
                })
                this.loadLessons()
            },
            onPickerCancel: data => {
                console.log(data);
            },
            onPickerSelect: data => {
                console.log(data);
            }
        });
        Picker.show();
    }

    loadLessons = () => {
        loadData({
            key:"lessons"
        }).then(data => {
            //console.log(data);
            const {week} = this.state;
            this.setState({
                classes: getWeekClassTable(data,week)
            })
        }).catch(err => {
            console.log(err)
        })
    }

    render() {
        return (
            <View>
                <View style={{ height: 50 }} style={styles.center}>
                    <TouchableOpacity onPress={this.timePick}>
                        <Text>
                            第{this.state.week}周
                        </Text>
                    </TouchableOpacity>
                </View>
                <View style={{ height: 40 }}>
                    <Grid>
                        <Row>
                            <Col style={styles.center} size={1}><Text style={{ fontSize: 12 }}>九月</Text></Col>
                            <Col style={styles.center} size={2}><Text>周一</Text></Col>
                            <Col style={styles.center} size={2}><Text>周二</Text></Col>
                            <Col style={styles.center} size={2}><Text>周三</Text></Col>
                            <Col style={styles.center} size={2}><Text>周四</Text></Col>
                            <Col style={styles.center} size={2}><Text>周五</Text></Col>
                            <Col style={styles.center} size={2}><Text>周六</Text></Col>
                            <Col style={styles.center} size={2}><Text>周日</Text></Col>
                        </Row>
                    </Grid>
                </View>
                <ScrollView>
                    <View style={{ height: 960, marginBottom: 85, }}>
                        <Grid>
                            <Col size={1}>
                                <Row style={styles.center} size={1}><Text>1</Text></Row>
                                <Row style={styles.center} size={1}><Text>2</Text></Row>
                                <Row style={styles.center} size={1}><Text>3</Text></Row>
                                <Row style={styles.center} size={1} ><Text>4</Text></Row>
                                <Row style={styles.center} size={1} ><Text>5</Text></Row>
                                <Row style={styles.center} size={1} ><Text>6</Text></Row>
                                <Row style={styles.center} size={1} ><Text>7</Text></Row>
                                <Row style={styles.center} size={1} ><Text>8</Text></Row>
                                <Row style={styles.center} size={1} ><Text>9</Text></Row>
                                <Row style={styles.center} size={1} ><Text>10</Text></Row>
                                <Row style={styles.center} size={1} ><Text>11</Text></Row>
                                <Row style={styles.center} size={1} ><Text>12</Text></Row>
                                <Row style={styles.center} size={1} ><Text>13</Text></Row>
                                <Row style={styles.center} size={1} ><Text>14</Text></Row>
                                <Row style={styles.center} size={1} ><Text>15</Text></Row>
                                <Row style={styles.center} size={1} ><Text>16</Text></Row>
                            </Col>
                            {
                                this.state.classes.map((weekday,index) => {
                                    return (
                                        <Col size={2} key={index} >
                                            {this.state.classes[index].map((lesson, index) => {
                                                return (
                                                    <Row size={lesson.span} key={index}>
                                                        {
                                                            lesson.name == null ? null :
                                                            <View style={styles.container}>
                                                                <Text style={styles.text}>{lesson.name}</Text>
                                                            </View>
                                                        }
                                                    </Row>
                                                )
                                            })}
                                        </Col>
                                    )
                                })
                            }
                        </Grid>
                    </View>
                </ScrollView>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'gainsboro',
    },
    center: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    text: {
        textAlign: 'center',
        fontSize: 14,
        lineHeight: 20,
    }
});

export default Classes