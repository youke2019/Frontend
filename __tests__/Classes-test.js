import Classes from '../src/pages/Classes'
import renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount, render, configure} from 'enzyme'
import Adapter from "enzyme-adapter-react-16/build"

configure({adapter: new Adapter()})

jest.mock('axios')
jest.mock("../src/utils/LocalStorage.js")
global.baseUrl = 'baseUrl'

test('renders classes page correctly', (done) => {
  const tree = renderer.create(
      <Classes />
  ).toJSON()

    const promise = new Promise(
        function (resolve, reject)
        {resolve('success')})
    promise
        .then(data=>{
            expect(tree).toMatchSnapshot();
            done()
        })
        .catch(err=>{console.log(err)})
});
