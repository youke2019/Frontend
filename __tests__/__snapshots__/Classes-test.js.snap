// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders classes page correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text>
        第
        1
        周
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 40,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          size={1}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            九月
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周一
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周二
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周三
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周四
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周五
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周六
          </Text>
        </View>
        <View
          size={2}
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            周日
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "height": 960,
            "marginBottom": 85,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            size={1}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                1
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                2
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                3
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                4
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                5
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                6
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                7
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                8
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                9
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                10
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                11
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                12
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                13
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                14
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                15
              </Text>
            </View>
            <View
              size={1}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                16
              </Text>
            </View>
          </View>
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
          <View
            size={2}
            style={
              Object {
                "flex": 2,
                "flexDirection": "column",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
